### PR TITLE
Azure-controllers metrics

### PR DESCRIPTION
**Pull request by charlesakalugwu**
This adds 4 new metrics for each of the azure-controllers:

- customeradmin-group-controller
- customeradmin-rolebinding-controller
- customeradmin-namespace-controller

The metrics are as follows:

- `azure_controllers_duration_seconds` (summary) with labels
  - controller
- `azure_controllers_errors_total` (counter) with labels
  - controller
- `azure_controllers_last_executed` (gauge) with labels
  - controller
- `azure_controllers_reconciliations_inflight` (gauge) with labels
  - controller

*Sample Scrape*
```
# HELP azure_controllers_duration_seconds The duration of azure controller runs.
# TYPE azure_controllers_duration_seconds summary
azure_controllers_duration_seconds{controller="customeradmin-group-controller",quantile="0.5"} 0.239014754
azure_controllers_duration_seconds{controller="customeradmin-group-controller",quantile="0.9"} 0.931348437
azure_controllers_duration_seconds{controller="customeradmin-group-controller",quantile="0.99"} 0.931348437
azure_controllers_duration_seconds_sum{controller="customeradmin-group-controller"} 1.879765005
azure_controllers_duration_seconds_count{controller="customeradmin-group-controller"} 5
azure_controllers_duration_seconds{controller="customeradmin-namespace-controller",quantile="0.5"} 1.092e-06
azure_controllers_duration_seconds{controller="customeradmin-namespace-controller",quantile="0.9"} 1.1511e-05
azure_controllers_duration_seconds{controller="customeradmin-namespace-controller",quantile="0.99"} 0.120646063
azure_controllers_duration_seconds_sum{controller="customeradmin-namespace-controller"} 0.20887089
azure_controllers_duration_seconds_count{controller="customeradmin-namespace-controller"} 21
azure_controllers_duration_seconds{controller="customeradmin-rolebinding-controller",quantile="0.5"} 1.393e-06
azure_controllers_duration_seconds{controller="customeradmin-rolebinding-controller",quantile="0.9"} 4.248e-06
azure_controllers_duration_seconds{controller="customeradmin-rolebinding-controller",quantile="0.99"} 6.0283e-05
azure_controllers_duration_seconds_sum{controller="customeradmin-rolebinding-controller"} 0.00027276
azure_controllers_duration_seconds_count{controller="customeradmin-rolebinding-controller"} 81
# HELP azure_controllers_errors_total Total number of errors.
# TYPE azure_controllers_errors_total counter
azure_controllers_errors_total{controller="customeradmin-group-controller"} 0
azure_controllers_errors_total{controller="customeradmin-namespace-controller"} 0
azure_controllers_errors_total{controller="customeradmin-rolebinding-controller"} 0
# HELP azure_controllers_last_executed The last time the azure controllers were run.
# TYPE azure_controllers_last_executed gauge
azure_controllers_last_executed{controller="customeradmin-group-controller"} 1.5617376255265226e+09
azure_controllers_last_executed{controller="customeradmin-namespace-controller"} 1.5617376185954323e+09
azure_controllers_last_executed{controller="customeradmin-rolebinding-controller"} 1.5617376188691056e+09
# HELP azure_controllers_reconciliations_inflight Number of azure controller reconciliations in progress.
# TYPE azure_controllers_reconciliations_inflight gauge
azure_controllers_reconciliations_inflight{controller="customeradmin-group-controller"} 0
azure_controllers_reconciliations_inflight{controller="customeradmin-namespace-controller"} 0
azure_controllers_reconciliations_inflight{controller="customeradmin-rolebinding-controller"} 0
```

/cc @jim-minter @kwoodson @Makdaam @mjudeikis  

```release-note
NONE
```

### CODE REVIEW

**Comment by charlesakalugwu** [comment]
upgrade-v4.4 flake #1759
upgrade-v5.1 no logs in CI
upgrade-v5.2 no logs in CI
e2e-create no logs in CI

**Comment by Makdaam** [comment]
/lgtm
metrics allow us to detect when controller is down, didn't run in a long time or has errors on recent runs.

**Comment by charlesakalugwu** [comment]
upgrade-v5.1 flake #1438 
/retest

File: pkg/controllers/customeradmin/group_controller.go [hunk 1]
```diff
@@ -97,8 +98,20 @@ func (r *reconcileGroup) load(path string) error {
 
 func (r *reconcileGroup) pollEvent(event.GenericEvent) bool {
 	r.log.Debug("AAD Group Reconciler (poll)..")
+
+	startTime := time.Now()
+	metricLabels := prometheus.Labels{"controller": "customeradmin-group-controller"}
+
+	azureControllersInFlightGauge.With(metricLabels).Inc()
```
Comment (kwoodson) [inline pkg/controllers/customeradmin/group_controller.go]: What do the increment and decrement calls do?

Comment (charlesakalugwu) [reply pkg/controllers/customeradmin/group_controller.go]: The Inc() and Dec() calls increment and decrement the value of the gauge respectively

**Pull request merged**